### ISSUE TITLE
Segfault on i386 test platform

### ISSUE THREAD

**Issue by stefanv** [scikit-image/scikit-image#2111]
See http://debomatic-i386.debian.net/distribution#unstable/skimage/0.12.3-1/buildlog

Python 2, skimage v0.12.3

**Comment by LocutusOfBorg** [comment]
skimage.morphology.tests.test_ccomp.TestConnectedComponents.test_return_num ... Segmentation fault (core dumped)

**Comment by LocutusOfBorg** [comment]
skimage.filters.rank.tests.test_rank.test_all ... FAIL
skimage.filters.rank.tests.test_rank.test_random_sizes ... Exception in thread Thread-10:
Traceback (most recent call last):
  File "/usr/lib/python2.7/threading.py", line 801, in __bootstrap_inner
    self.run()
  File "/usr/lib/python2.7/threading.py", line 754, in run
    self.__target(*self.__args, **self.__kwargs)
  File "/build/skimage-0.12.3/debian/tmp/usr/lib/python2.7/dist-packages/skimage/filters/rank/tests/test_rank.py", line 83, in check_all
    rank.entropy(image, selem))
  File "/usr/lib/python2.7/dist-packages/numpy/testing/utils.py", line 316, in assert_equal
    return assert_array_equal(actual, desired, err_msg, verbose)
  File "/usr/lib/python2.7/dist-packages/numpy/testing/utils.py", line 807, in assert_array_equal
    verbose=verbose, header='Arrays are not equal')
  File "/usr/lib/python2.7/dist-packages/numpy/testing/utils.py", line 733, in assert_array_compare
    raise AssertionError(msg)
AssertionError:
Arrays are not equal

**Comment by LocutusOfBorg** [comment]
do you need more info?
they are reproducible with pbuilder-dist sid i386 create
pbuilder-dist sid i386 build filename.dsc or pbuilder-dist sid i386 login

**Comment by LocutusOfBorg** [comment]
this happened in the official Debian builds too
https://buildd.debian.org/status/package.php?p=skimage&suite=unstable
armhf i386 hppa and probably more archs have this issue.

**Comment by LocutusOfBorg** [comment]
I see many segmentation faults:
"test_clear_border_non_binary"
and a lot of test_ccomp.py failures, all related to the same issue
test_return_num
the 
assert_array_equal(label(x, background=-1, return_num=True)[1], 4)
^^^ is segfaulting
and a test failure
```
======================================================================
FAIL: skimage.feature.tests.test_orb.test_descriptor_orb
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/usr/lib/python2.7/dist-packages/nose/case.py", line 197, in runTest
    self.test(*self.arg)
  File "/skimage-0.12.3/debian/tmp/usr/lib/python2.7/dist-packages/skimage/feature/tests/test_orb.py", line 99, in test_descriptor_orb
    detector_extractor.descriptors[100:120, 10:20])
  File "/usr/lib/python2.7/dist-packages/numpy/testing/utils.py", line 320, in assert_equal
    return assert_array_equal(actual, desired, err_msg, verbose)
  File "/usr/lib/python2.7/dist-packages/numpy/testing/utils.py", line 811, in assert_array_equal
    verbose=verbose, header='Arrays are not equal')
  File "/usr/lib/python2.7/dist-packages/numpy/testing/utils.py", line 737, in assert_array_compare
    raise AssertionError(msg)
AssertionError: 
Arrays are not equal

(mismatch 46.5%)
 x: array([[False,  True,  True,  True, False,  True, False,  True, False,
         True],
       [ True,  True,  True, False, False,  True, False, False,  True,...
 y: array([[ True, False, False, False, False, False, False, False,  True,
         True],
       [False,  True,  True,  True, False,  True, False,  True, False,...

======================================================================
FAIL: skimage.filters.rank.tests.test_rank.test_all
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/usr/lib/python2.7/dist-packages/nose/case.py", line 197, in runTest
    self.test(*self.arg)
  File "/skimage-0.12.3/debian/tmp/usr/lib/python2.7/dist-packages/skimage/filters/rank/tests/test_rank.py", line 16, in test_all
    check_all()
  File "/skimage-0.12.3/debian/tmp/usr/lib/python2.7/dist-packages/skimage/_shared/testing.py", line 232, in inner
    result = func(*args, **kwargs)
  File "/skimage-0.12.3/debian/tmp/usr/lib/python2.7/dist-packages/skimage/filters/rank/tests/test_rank.py", line 83, in check_all
    rank.entropy(image, selem))
  File "/usr/lib/python2.7/dist-packages/numpy/testing/utils.py", line 320, in assert_equal
    return assert_array_equal(actual, desired, err_msg, verbose)
  File "/usr/lib/python2.7/dist-packages/numpy/testing/utils.py", line 811, in assert_array_equal
    verbose=verbose, header='Arrays are not equal')
  File "/usr/lib/python2.7/dist-packages/numpy/testing/utils.py", line 737, in assert_array_compare
    raise AssertionError(msg)
AssertionError: 
Arrays are not equal

(mismatch 1.28%)
 x: array([[ 1.584963,  2.      ,  2.      ,  2.      ,  2.      ,  2.      ,
         2.      ,  2.      ,  2.      ,  2.      ,  2.      ,  2.      ,
         2.      ,  2.      ,  2.      ,  2.      ,  2.      ,  2.      ,...
 y: array([[ 1.584963,  2.      ,  2.      ,  2.      ,  2.      ,  2.      ,
         2.      ,  2.      ,  2.      ,  2.      ,  2.      ,  2.      ,
         2.      ,  2.      ,  2.      ,  2.      ,  2.      ,  2.      ,...

----------------------------------------------------------------------
```

**Comment by tfmoraes** [comment]
More about this error here https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=840589

**Comment by matthew-brett** [comment]
Thanks for pointing out that Debian analysis, which has found a bug in the Cython code.  I make a more specific issue for the error : #2369

**Comment by soupault** [comment]
Fixed via #2369. Other issues are covered by #2528, #2529.
